### ISSUE TITLE
Generalize SUAVE Execution Namespace

### ISSUE THREAD

**Issue by dmarzzz** [flashbots/suave-specs#79]
As it stand SUAVE is only configured to work with Dencun Compatible L1s, and only one at a time. How can we generalize the execution namespace and SUAVE to enable routing execution calls to nodes on any chain which support the execution namespace? One thing to note is that eventually we will be able to send the JSON RPC request for buildEthBlock using the `doHttpRequest` precompile, therefore the smart contract can craft the payload in whatever way, but the user will still need to pick a chain to send to and know what chains are available to the kettle

**Comment by jinmel** [comment]
https://gist.github.com/jinmel/54fd522db8bd1ef757b797588adc7310

**Comment by jinmel** [comment]
I thought it should be a the kettle operator running the rpc to keep everything under TEE. So setting arbitrary urls in contract should be forbidden. I think the dns approach sounds more extensible than chain id because now you can have multiple rpc endpoint per chain. i.e.) goerli-1, goerli-2. not sure if it is necessary to have multiple endpoints.

**Comment by dmarzzz** [comment]
Yeah there is a friction here good call out @jinmel . when reading a suave contract we need to be able to guarantee that the URL someone is using to simulate is the kettle's domain service (or another kettle's domain service). I think with what @ferranbt recommended this is possible.

 also I like the idea of multiple endpoints per chain, sims can likely arbitrarily scale so this will be good to support, the main thing we will need to abstract away though in the multi domain setting is making sure sessions with "connection IDs" continue to use the same endpoint. Perhaps we can do this as follow on work and just assume 1 endpoint per chain for now.

**Comment by jinmel** [comment]
in your scenario @ferranbt, does the kettle operator set the whitelisted url endpoints that suapps can talk with? I am mildly concerned about the security of user setting arbitrary raw url. This is literally one loop hole where everything can be leaked out.

**Comment by jinmel** [comment]
@dmarzzz do you mean the builder sessions? I think that can be easily managed in the next PR.

**Comment by ferranbt** [comment]
> in your scenario @ferranbt, does the kettle operator set the whitelisted url endpoints that suapps can talk with? I am mildly concerned about the security of user setting arbitrary raw url. This is literally one loop hole where everything can be leaked out.

Yes, it is the kettle that whitelists the addresses. Of course, if the address is set on this dns registry, it is automatically whitelisted.
